### ISSUE TITLE
Ref to external JSON Schema causes error "😱 Could not render Mt"

### ISSUE THREAD

**Issue by jasny** [swagger-api/swagger-ui#8895]
### Q&A (please complete the following information)
 - OS: Ubuntu Linux
 - Browser: chrome
 - Version: 114
 - Method of installation: NPM via @nestjs/swagger
 - Swagger-UI version: 4.18.2
 - Swagger/OpenAPI version: 3.0.0

### Content & configuration

Example Swagger/OpenAPI definition:
```yaml
openapi: 3.0.0
paths:
  /scenarios:
    post:
      operationId: ScenarioController_store
      summary: Store a scenario
      parameters: []
      requestBody:
        required: true
        content:
          application/json: &ref_0
            schema:
              $ref: https://schemas.letsflow.io/v1.0.0/scenario
          application/yaml: *ref_0
      responses:
        '201':
          description: Created
      tags: *ref_1
```

### Describe the bug you're encountering
Using this schema causes a javascript error.

```
react-dom.production.min.js:216 TypeError: Cannot create property 'type' on boolean 'true'
    at F (fn.js:230:14)
    at F.$ (fn.js:363:25)
    at F (fn.js:483:7)
    at F (fn.js:502:36)
    at F.$ (fn.js:363:25)
    at F (fn.js:483:7)
    at q (fn.js:630:3)
    at n (memoize.js:62:23)
    at He (utils.js:634:15)
    at Ke (utils.js:679:10)
```

### To reproduce...

* Open `POST scenarios` endpoint.
* Error `😱 Could not render Mt, see the console.` is displayed.

### Expected behavior
https://schemas.letsflow.io/v1.0.0/scenario is a valid schema. It should not produce an error.

### Screenshots
![Screenshot from 2023-06-07 17-34-41](https://github.com/swagger-api/swagger-ui/assets/100821/e252d99b-c33a-47b2-9e8d-99c075fbf201)

**Comment by qqilihq** [comment]
We experience this as well - also happens with `5.0.0-alpha.15`

**Comment by j-ibarra** [comment]
`😱 Could not render Mn, see the console.`

**Comment by qqilihq** [comment]
Resolved with 5.0.0
